hs_access_control/models/community.py

```diff
@@ -33,35 +33,26 @@ def owners(self):
                                    u2ucp__privilege=PrivilegeCodes.OWNER)
 
     @property
-    def public_resource_list(self):
+    def public_resources(self):
         """
         prepare a list of everything that gets displayed about each resource in a community.
         """
         # TODO: consider adding GenericRelation to expose reverse querying of metadata field.
         # TODO: This would enable fast querying of first author.
         # TODO: The side-effect of this is enabling deletion cascade, which shouldn't do anything.
-        return BaseResource.objects.filter(r2grp__group__g2gcp__community=self,
-                                           r2grp__group__gaccess__active=True)\
-                                   .filter(Q(raccess__public=True) |
-                                           Q(raccess__published=True) |
-                                           Q(raccess__discoverable=True))\
-                                   .annotate(group_id=F("r2grp__group__id"),
-                                             group_name=F("r2grp__group__name"),
-                                             resource_id=F("short_id"),
-                                             resource_title=F("title"),
-                                             published=F("raccess__published"),
-                                             public=F("raccess__public"),
-                                             discoverable=F("raccess__discoverable"))\
-                                   .values("group_id",
-                                           "group_name",
-                                           "resource_id",
-                                           "resource_title",
-                                           "resource_type",
-                                           "discoverable",
-                                           "public",
-                                           "published",
-                                           "created",
-                                           "updated")
+        res = BaseResource.objects.filter(r2grp__group__g2gcp__community=self,
+                                          r2grp__group__gaccess__active=True)\
+                                  .filter(Q(raccess__public=True) |
+                                          Q(raccess__published=True) |
+                                          Q(raccess__discoverable=True))\
+                                  .annotate(group_name=F("r2grp__group__name"),
+                                            group_id=F("r2grp__group__id"))
+
+        res = res.only('title', 'resource_type', 'created', 'updated')
+        res = res.prefetch_related("content_object___title",
```

Comment:
I had to roll back this prefetch strategy in optimization of my resources page. I discovered that this prefetch of 'content-object' is problematic when trying to resolve the filtering of resources by creator on my resources page. It seems prefect of 'content_object' is capable of prefetching content_object of only one ContentType ('e.g., CoreMetadata). If we had all our resources as Composite Resource this would work since all the 'content_object's are of type CoreMetaData - single ContentType.

Reply:
Here is Django official documentation:

>  It also supports prefetching of GenericRelation and GenericForeignKey, however, it must be restricted to a homogeneous set of results. For example, prefetching objects referenced by a GenericForeignKey is only supported if the query is restricted to one ContentType.

https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.prefetch_related

Reply:
Changed to utilize the new pattern you suggested. It seems to work fine.

Reply:
@pkdash I implemented the new polymorphic pattern and it seems to work properly.

Reply:
@alvacouch Is the polymorphic pattern you implemented in this PR?

Reply:
Yes, it is in the `hs_access_control/models/community.py:Community:public_resources` method.